backends/dpdk/dpdkArch.h

```diff
@@ -894,26 +902,59 @@ class SplitActionSelectorTable : public SplitP4TableCommon {
  */
 class SplitActionProfileTable : public SplitP4TableCommon {
  public:
-    SplitActionProfileTable(P4::ReferenceMap* refMap, P4::TypeMap* typeMap,
-                            DpdkProgramStructure* structure)
-        : SplitP4TableCommon(refMap, typeMap, structure) {
+    SplitActionProfileTable(
+        P4::ReferenceMap* refMap, P4::TypeMap* typeMap, DpdkProgramStructure* structure,
+        std::map<cstring, std::vector<std::tuple<cstring, IR::Constant*>>>& actionCaseMap)
+        : SplitP4TableCommon(refMap, typeMap, structure, actionCaseMap) {
         implementation = TableImplementation::ACTION_PROFILE;
     }
     const IR::Node* postorder(IR::P4Table* tbl) override;
 };
 
+class UpdateActionForSwitch : public Transform {
```

Comment:
I also try to document the passes I am building.

Reply:
Sure, will do.